pkg/awsutils/awsutils.go

```diff
@@ -1225,7 +1288,15 @@ func (cache *EC2InstanceMetadataCache) GetVPCIPv4CIDR() string {
 
 // GetVPCIPv4CIDRs returns VPC CIDRs
 func (cache *EC2InstanceMetadataCache) GetVPCIPv4CIDRs() []*string {
-	return cache.vpcIPv4CIDRs
+	cache.vpcIPv4CIDRs.RLock()
+	defer cache.vpcIPv4CIDRs.RUnlock()
+	var vpcIPv4CIDRslice []*string
+	for vpcIPv4CIDR := range cache.vpcIPv4CIDRs.data{
+		//In go, the loop iterator variable is a single variable that takes different values in each loop iteration.
+		tmpVar := vpcIPv4CIDR
+		vpcIPv4CIDRslice = append(vpcIPv4CIDRslice, aws.String(tmpVar))
+	}
+	return vpcIPv4CIDRslice
```

Comment:
and then the above would need to change to just this:

```go
return cache.vpcIPv4CIDRs.AWSStrings()
```

since you would place the RLock()/RUnlock() within the `StringSet.AWSStrings()` implementation